File: 프로그래머스/2/42584. 주식가격/주식가격.py
"""
입력[2, 2, 3, 1, 5]
출력[3, 2, 1, 1, 0]
"""

def solution(prices):
    n = len(prices)
    stack = []
    answer = [-1]*(n)
    stack=[0]
    for i in range(1,n):
        price = prices[i]
        while stack and prices[stack[-1]]>prices[i]:
            x = stack.pop()
            answer[x] = i-x
        stack.append(i)
            
    while stack:
        x = stack.pop()
        answer[x] = i-x

    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const N = prices.length
    const answer = Array.from({length:N}).fill(0);
    const stack = [0]
    for(let i=1;i<N;i++){
        while(stack.length>0 && prices[stack[stack.length-1]]>prices[i]){
            j=stack.pop()
            answer[j] = i-j
        }
        stack.push(i)
        
    }
    while(stack.length>0){
        const j = stack.pop()
        answer[j] = N-1-j
    }

    console.log(answer, stack)

    return answer;
}